app/connectors/responses/DeleteStateBenefitDetailOverrideResponse.scala

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package connectors.responses
+
+import connectors.Parser
+import connectors.errors.ApiError
+import play.api.http.Status._
+import uk.gov.hmrc.http.{HttpReads, HttpResponse}
+
+case class DeleteStateBenefitDetailOverrideResponse(httpResponse: HttpResponse, result: Either[ApiError, Unit])
+
+object DeleteStateBenefitDetailOverrideResponse {
+
+  implicit val deleteStateBenefitDetailOverrideResponseReads: HttpReads[DeleteStateBenefitDetailOverrideResponse] =
+    new HttpReads[DeleteStateBenefitDetailOverrideResponse] with Parser {
+      override protected[connectors] val parserName: String = this.getClass.getSimpleName
+
+      override def read(method: String, url: String, response: HttpResponse): DeleteStateBenefitDetailOverrideResponse = response.status match {
+        case NO_CONTENT => DeleteStateBenefitDetailOverrideResponse(response, Right(()))
+        case BAD_REQUEST | NOT_FOUND | UNPROCESSABLE_ENTITY | INTERNAL_SERVER_ERROR | SERVICE_UNAVAILABLE =>
+          DeleteStateBenefitDetailOverrideResponse(response, handleError(response, response.status))
+        case _ => DeleteStateBenefitDetailOverrideResponse(response, handleError(response, INTERNAL_SERVER_ERROR))
```

Comment:
All cases of this match expression return `DeleteStateBenefitDetailOverrideResponse(response, `. You could put this common part outside the match so the match just returns the 2nd argument (the result) to `DeleteStateBenefitDetailOverrideResponse`

Reply:
This is true, however, if I did that I would have had to extract this as a variable and make the method a bit less readable. At the moment the reat definition does not need a body. If I did what you suggested I would have to add that.

Reply:
How about instead of extracting it into a variable, making it point-free like this? I think this reads nicely and makes it clearer what the difference is between the cases. It also means each case fits on one line
```scala
      override def read(method: String, url: String, response: HttpResponse): DeleteStateBenefitDetailOverrideResponse =
        DeleteStateBenefitDetailOverrideResponse(response,
          response.status match {
            case NO_CONTENT => Right(())
            case BAD_REQUEST | NOT_FOUND | UNPROCESSABLE_ENTITY | INTERNAL_SERVER_ERROR | SERVICE_UNAVAILABLE => handleError(response, response.status)
            case _ => handleError(response, INTERNAL_SERVER_ERROR)
          }
        )

Reply:
Is it more readable?

Reply:
Maybe, but it's subjective of course - here's the original version, along with the point-free version, and a version where the result's been extracted as a variable:
```scala
override def read(method: String, url: String, response: HttpResponse): DeleteStateBenefitDetailOverrideResponse = response.status match {
  case NO_CONTENT => DeleteStateBenefitDetailOverrideResponse(response, Right(()))
  case BAD_REQUEST | NOT_FOUND | UNPROCESSABLE_ENTITY | INTERNAL_SERVER_ERROR | SERVICE_UNAVAILABLE =>
    DeleteStateBenefitDetailOverrideResponse(response, handleError(response, response.status))
  case _ => DeleteStateBenefitDetailOverrideResponse(response, handleError(response, INTERNAL_SERVER_ERROR))
}

override def read2(method: String, url: String, response: HttpResponse): DeleteStateBenefitDetailOverrideResponse =
  DeleteStateBenefitDetailOverrideResponse(response,
    response.status match {
      case NO_CONTENT => Right(())
      case BAD_REQUEST | NOT_FOUND | UNPROCESSABLE_ENTITY | INTERNAL_SERVER_ERROR | SERVICE_UNAVAILABLE => handleError(response, response.status)
      case _ => handleError(response, INTERNAL_SERVER_ERROR)
    }
  )

override def read3(method: String, url: String, response: HttpResponse): DeleteStateBenefitDetailOverrideResponse = {
  val result = response.status match {
    case NO_CONTENT => Right(())
    case BAD_REQUEST | NOT_FOUND | UNPROCESSABLE_ENTITY | INTERNAL_SERVER_ERROR | SERVICE_UNAVAILABLE => handleError(response, response.status)
    case _ => handleError(response, INTERNAL_SERVER_ERROR)
  }
  DeleteStateBenefitDetailOverrideResponse(response, result)
}
```
What do you think? I prefer versions 2 and 3 over 1.

Reply:
Which version has the lowest number of lines? The difference between is not that big, I don't see an improvement from the first version.
Lets talk about it offline. This is not really a good tool for conversation :)